src/radical/entk/execman/rp/task_processor.py

```diff
@@ -153,14 +153,26 @@ def resolve_arguments(args, placeholders):
 
 # ------------------------------------------------------------------------------
 #
-def resolve_tags(tag, parent_pipeline_name, placeholders):
+def resolve_tags(tags, parent_pipeline_name, placeholders):
+
+    # entk only handles co_location tags.  If tags are given as strings, they
+    # get translated into `{'colocation': '<tag>'}`.  Tags passed as dictionaies
+    # are checked to conform with above form.
+    #
+    # In both cases, the tag string is expanded with the given placeholders.
+
+    if not tags:
+        return
+
+    colo_tag = tags['colocate']
```

Comment:
Should the string type check go here maybe (the one mentioned in the comment above)?

Reply:
Resolved during task definition